acceptance: localcluster nodes sometimes fail to get a NodeID assigned to them

The following tests appear to have failed:

[#40720](https://teamcity.cockroachdb.com/viewLog.html?buildId=40720):

```
--- FAIL: acceptance/TestGossipPeerings (462.293s)
Test ended in panic.
I161102 19:31:58.720205 600 acceptance/cluster/localcluster.go:290  creating docker network with name: cockroachdb_acceptance-102ad794
I161102 19:31:58.824013 600 acceptance/cluster/localcluster.go:322  Initializing Cluster AdHoc 3x1:
{"name":"AdHoc 3x1","nodes":[{"count":3,"stores":[{"count":1,"max_ranges":0}]}],"duration":5000000000}
I161102 19:31:58.829739 600 acceptance/cluster/docker.go:107  ImagePull cockroachdb/builder:20161024-143446 already exists
I161102 19:32:00.615904 600 acceptance/cluster/localcluster.go:641  creating certs (1024bit) in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster.certs.985011928
I161102 19:32:00.671832 600 acceptance/cluster/localcluster.go:425  creating docker container with name: roach-102ad794-0
I161102 19:32:01.312168 600 acceptance/cluster/localcluster.go:536  *** started roach-102ad794-0 ***
  ui:        https://127.0.0.1:33157
  trace:     https://127.0.0.1:33157/debug/requests
  logs:      /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestGossipPeerings-102ad794/roach-102ad794-0/cockroach.INFO
  pprof:     docker exec -it b1741 /bin/bash -c 'go tool pprof /cockroach <(wget --no-check-certificate -qO- https://$(hostname):8080/debug/pprof/heap)'
  cockroach: [start --ca-cert=/certs/ca.crt --cert=/certs/node.crt --key=/certs/node.key --host=roach-102ad794-0 --alsologtostderr=INFO --verbosity=1 --store=path=/data1.0 --log-dir=/logs/roach-102ad794-0 --logtostderr=false]

  cli-env:   COCKROACH_INSECURE=false COCKROACH_CA_CERT=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster.certs.985011928/ca.crt COCKROACH_CERT=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster.certs.985011928/node.crt COCKROACH_KEY=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster.certs.985011928/node.key COCKROACH_HOST=127.0.0.1 COCKROACH_PORT=33157
I161102 19:32:01.312261 600 acceptance/cluster/localcluster.go:425  creating docker container with name: roach-102ad794-1
I161102 19:32:01.963417 600 acceptance/cluster/localcluster.go:536  *** started roach-102ad794-1 ***
  ui:        https://127.0.0.1:33159
  trace:     https://127.0.0.1:33159/debug/requests
  logs:      /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestGossipPeerings-102ad794/roach-102ad794-1/cockroach.INFO
  pprof:     docker exec -it 32608 /bin/bash -c 'go tool pprof /cockroach <(wget --no-check-certificate -qO- https://$(hostname):8080/debug/pprof/heap)'
  cockroach: [start --ca-cert=/certs/ca.crt --cert=/certs/node.crt --key=/certs/node.key --host=roach-102ad794-1 --alsologtostderr=INFO --verbosity=1 --store=path=/data2.0 --join=roach-102ad794-0:26257 --log-dir=/logs/roach-102ad794-1 --logtostderr=false]

  cli-env:   COCKROACH_INSECURE=false COCKROACH_CA_CERT=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster.certs.985011928/ca.crt COCKROACH_CERT=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster.certs.985011928/node.crt COCKROACH_KEY=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster.certs.985011928/node.key COCKROACH_HOST=127.0.0.1 COCKROACH_PORT=33159
I161102 19:32:01.963507 600 acceptance/cluster/localcluster.go:425  creating docker container with name: roach-102ad794-2
I161102 19:32:02.611323 600 acceptance/cluster/localcluster.go:536  *** started roach-102ad794-2 ***
  ui:        https://127.0.0.1:33161
  trace:     https://127.0.0.1:33161/debug/requests
  logs:      /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestGossipPeerings-102ad794/roach-102ad794-2/cockroach.INFO
  pprof:     docker exec -it 10e43 /bin/bash -c 'go tool pprof /cockroach <(wget --no-check-certificate -qO- https://$(hostname):8080/debug/pprof/heap)'
  cockroach: [start --ca-cert=/certs/ca.crt --cert=/certs/node.crt --key=/certs/node.key --host=roach-102ad794-2 --alsologtostderr=INFO --verbosity=1 --store=path=/data3.0 --join=roach-102ad794-0:26257 --log-dir=/logs/roach-102ad794-2 --logtostderr=false]

  cli-env:   COCKROACH_INSECURE=false COCKROACH_CA_CERT=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster.certs.985011928/ca.crt COCKROACH_CERT=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster.certs.985011928/node.crt COCKROACH_KEY=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster.certs.985011928/node.key COCKROACH_HOST=127.0.0.1 COCKROACH_PORT=33161
I161102 19:32:02.611341 600 acceptance/replication_test.go:55  waiting for first range to have 3 replicas
I161102 19:32:03.613792 600 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161102 19:32:04.616604 600 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161102 19:32:05.619039 600 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161102 19:32:06.621365 600 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161102 19:32:07.624011 600 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161102 19:32:08.629248 600 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161102 19:32:09.633077 600 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
panic: test timed out after 10m0s

goroutine 666 [running]:
panic(0x13d3280, 0xc4203efd80)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:918 +0x10b
created by time.goFunc
	/usr/local/go/src/time/sleep.go:154 +0x44

goroutine 1 [chan receive, 7 minutes]:
testing.(*T).Run(0xc420424180, 0x156811a, 0x12, 0x16ac130, 0xc42028dc01)
	/usr/local/go/src/testing/testing.go:647 +0x316
testing.RunTests.func1(0xc420424180)
	/usr/local/go/src/testing/testing.go:793 +0x6d
testing.tRunner(0xc420424180, 0xc420044da8)
	/usr/local/go/src/testing/testing.go:610 +0x81
testing.RunTests(0x16aea68, 0x1e8dfe0, 0x24, 0x24, 0xc42028ddf0)
	/usr/local/go/src/testing/testing.go:799 +0x2f5
testing.(*M).Run(0xc420044ef8, 0x16ac1e0)
	/usr/local/go/src/testing/testing.go:743 +0x85
github.com/cockroachdb/cockroach/pkg/acceptance.runTests(0xc420044ef8)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_test.go:138 +0x48
github.com/cockroachdb/cockroach/pkg/acceptance.TestMain(0xc420044ef8)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/main_test.go:38 +0x45
main.main()
	github.com/cockroachdb/cockroach/pkg/acceptance/_test/_testmain.go:122 +0xc6

goroutine 17 [syscall, 9 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 7 [syscall, 9 minutes]:
os/signal.signal_recv(0xc42002ffa8)
	/usr/local/go/src/runtime/sigqueue.go:116 +0x157
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 6 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).flushDaemon(0x21f4b60)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1001 +0x77
created by github.com/cockroachdb/cockroach/pkg/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:579 +0x95

goroutine 371 [select, 8 minutes]:
net/http.(*persistConn).writeLoop(0xc420502400)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 21 [chan receive, 9 minutes]:
github.com/cockroachdb/cockroach/pkg/acceptance.runTests.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_test.go:129 +0xb6
created by github.com/cockroachdb/cockroach/pkg/acceptance.runTests
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_test.go:137 +0x3a

goroutine 600 [select, 7 minutes]:
google.golang.org/grpc/transport.(*Stream).Header(0xc42028c4b0, 0x16ae8a0, 0xc4205711e8, 0x1eb00c0)
	/go/src/google.golang.org/grpc/transport/transport.go:239 +0x299
google.golang.org/grpc.recvResponse(0x1eaa040, 0x222dc30, 0x0, 0x0, 0x0, 0x0, 0x1e9db80, 0xc420380a80, 0x0, 0x0, ...)
	/go/src/google.golang.org/grpc/call.go:62 +0xad
google.golang.org/grpc.Invoke(0x7fa82bf64208, 0xc420014558, 0x1576756, 0x21, 0x15412e0, 0xc4204d6380, 0x14e1480, 0xc420326600, 0xc4204eec00, 0x0, ...)
	/go/src/google.golang.org/grpc/call.go:202 +0x8a1
github.com/cockroachdb/cockroach/pkg/roachpb.(*externalClient).Batch(0xc420144210, 0x7fa82bf64208, 0xc420014558, 0xc4204d6380, 0x0, 0x0, 0x0, 0xc420571500, 0x72b3f8, 0xc42027c870)
	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:1562 +0xd2
github.com/cockroachdb/cockroach/pkg/internal/client.sender.Send(0x1e97980, 0xc420144210, 0x7fa82bf64208, 0xc420014558, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/rpc_sender.go:48 +0xbf
github.com/cockroachdb/cockroach/pkg/internal/client.(*sender).Send(0xc4203c9a60, 0x7fa82bf64208, 0xc420014558, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	<autogenerated>:4 +0xb0
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).send(0xc420284640, 0x7fa82bf64208, 0xc420014558, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:511 +0xbf
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Run.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:445 +0x6e
github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill(0xc420552100, 0xc4201c4000, 0x0, 0x1eb0b40)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:416 +0x168
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Run(0xc420284640, 0x7fa82bf64208, 0xc420014558, 0xc4201c4000, 0x8, 0x18)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:447 +0xac
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Get(0xc420284640, 0x7fa82bf64208, 0xc420014558, 0x14b8e40, 0xc4205520e0, 0xc4205520e0, 0x18, 0x18, 0x1, 0xc4205520e0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:214 +0x9a
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).GetProto(0xc420284640, 0x7fa82bf64208, 0xc420014558, 0x14b8e40, 0xc4205520e0, 0x1ea40c0, 0xc4203265a0, 0xc4205719e0, 0x60033c)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:222 +0x69
github.com/cockroachdb/cockroach/pkg/acceptance.countRangeReplicas(0xc420284640, 0xc420219b00, 0x0, 0xc420284601)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/replication_test.go:36 +0x157
github.com/cockroachdb/cockroach/pkg/acceptance.checkRangeReplication.func1(0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/replication_test.go:71 +0x1f4
github.com/cockroachdb/cockroach/pkg/util.RetryForDuration(0xa7a358200, 0xc420571ce0, 0x2, 0x1583329)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/testing.go:129 +0xcd
github.com/cockroachdb/cockroach/pkg/util.SucceedsSoonDepth(0x1, 0x1ead480, 0xc4204243c0, 0xc420571ce0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/testing.go:115 +0x41
github.com/cockroachdb/cockroach/pkg/util.SucceedsSoon(0x1ead480, 0xc4204243c0, 0xc420571ce0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/testing.go:109 +0x48
github.com/cockroachdb/cockroach/pkg/acceptance.checkRangeReplication(0xc4204243c0, 0x1eb1be0, 0xc420336200, 0x4a817c800)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/replication_test.go:83 +0x2d3
github.com/cockroachdb/cockroach/pkg/acceptance.StartCluster(0xc4204243c0, 0xc420515750, 0x9, 0xc42026a320, 0x1, 0x1, 0x12a05f200, 0x1eb1be0, 0xc420336200)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_test.go:345 +0x188
github.com/cockroachdb/cockroach/pkg/acceptance.runTestOnConfigs.func1(0xc4204243c0, 0xc420571ec8, 0x16ac228)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_test.go:312 +0x4e
github.com/cockroachdb/cockroach/pkg/acceptance.runTestOnConfigs(0xc4204243c0, 0x16ac228)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_test.go:315 +0xe2
github.com/cockroachdb/cockroach/pkg/acceptance.TestGossipPeerings(0xc4204243c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/gossip_peerings_test.go:106 +0x37
testing.tRunner(0xc4204243c0, 0x16ac130)
	/usr/local/go/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:646 +0x2ec

goroutine 9 [select, 9 minutes, locked to thread]:
runtime.gopark(0x16af320, 0x0, 0x155b925, 0x6, 0x18, 0x2)
	/usr/local/go/src/runtime/proc.go:259 +0x13a
runtime.selectgoImpl(0xc420030f30, 0x0, 0x18)
	/usr/local/go/src/runtime/select.go:423 +0x11d9
runtime.selectgo(0xc420030f30)
	/usr/local/go/src/runtime/select.go:238 +0x1c
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal1_unix.go:304 +0x2f3
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 24 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7fa82bf1d0b0, 0x72, 0x4)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420463720, 0x72, 0xc42003e9d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420463720, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc4204636c0, 0xc420469000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc420144140, 0xc420469000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc420146a00, 0xc420469000, 0x1000, 0x1000, 0x30, 0xc42003eb58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc42014cc60)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc42014cc60, 0x1, 0x0, 0x1, 0x0, 0xc42014c4e0, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc420146a00)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 25 [select, 9 minutes]:
net/http.(*persistConn).writeLoop(0xc420146a00)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 244 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7fa82bf1cff0, 0x72, 0xa)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc42036a220, 0x72, 0xc4200409d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc42036a220, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc42036a1c0, 0xc420480000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc420510040, 0xc420480000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc4204ee400, 0xc420480000, 0x1000, 0x1000, 0x30, 0xc420040b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc42031c660)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc42031c660, 0x1, 0x0, 0x1, 0x0, 0xc42031cde0, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc4204ee400)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 47 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7fa82bf1cf30, 0x72, 0x6)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc4203d14f0, 0x72, 0xc4200419d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc4203d14f0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc4203d1490, 0xc420485000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc420144128, 0xc420485000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc420502200, 0xc420485000, 0x1000, 0x1000, 0x30, 0xc420041b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc420264fc0)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc420264fc0, 0x1, 0x0, 0x1, 0x0, 0xc42014ccc0, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc420502200)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 48 [select, 9 minutes]:
net/http.(*persistConn).writeLoop(0xc420502200)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 434 [select, 8 minutes]:
net/http.(*persistConn).writeLoop(0xc420502100)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 255 [select, 9 minutes]:
net/http.(*persistConn).writeLoop(0xc4204ee600)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 245 [select, 9 minutes]:
net/http.(*persistConn).writeLoop(0xc4204ee400)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 208 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7fa82bf1c9f0, 0x72, 0xe)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc42029a8b0, 0x72, 0xc4205599d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc42029a8b0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc42029a850, 0xc42046a000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc42012a200, 0xc42046a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc4204ee700, 0xc42046a000, 0x1000, 0x1000, 0x30, 0xc420559b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc42031d9e0)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc42031d9e0, 0x1, 0x0, 0x1, 0x0, 0xc420166a80, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc4204ee700)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 185 [select, 9 minutes]:
net/http.(*persistConn).writeLoop(0xc4204ee300)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 171 [select, 9 minutes]:
net/http.(*persistConn).writeLoop(0xc420336100)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 209 [select, 9 minutes]:
net/http.(*persistConn).writeLoop(0xc4204ee700)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 514 [select, 8 minutes]:
net/http.(*persistConn).writeLoop(0xc4204eed00)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 353 [IO wait, 8 minutes]:
net.runtime_pollWait(0x7fa82bf1c4b0, 0x72, 0x15)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc4202780d0, 0x72, 0xc4205b59d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc4202780d0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc420278070, 0xc42027a000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc420510018, 0xc42027a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc420502100, 0xc42027a000, 0x1000, 0x1000, 0x30, 0xc4205b5b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc4200542a0)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc4200542a0, 0x1, 0x0, 0x1, 0x0, 0xc4202643c0, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc420502100)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 266 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7fa82bf1c870, 0x72, 0x10)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc4201698e0, 0x72, 0xc4202c79d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc4201698e0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc420169880, 0xc4200f8000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc420510148, 0xc4200f8000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc420502900, 0xc4200f8000, 0x1000, 0x1000, 0x30, 0xc4202c7b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc420326a20)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc420326a20, 0x1, 0x0, 0x1, 0x0, 0xc420427080, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc420502900)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 128 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7fa82bf1ccf0, 0x72, 0x7)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420010450, 0x72, 0xc4201389d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420010450, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc4200103f0, 0xc42029f000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc42012a188, 0xc42029f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc42000b400, 0xc42029f000, 0x1000, 0x1000, 0x30, 0xc420138b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc42028a840)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc42028a840, 0x1, 0x0, 0x1, 0x0, 0xc42050e000, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc42000b400)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 198 [chan receive, 9 minutes]:
database/sql.(*DB).connectionOpener(0xc4203a02c0)
	/usr/local/go/src/database/sql/sql.go:730 +0x4a
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x1e9

goroutine 170 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7fa82bf1ce70, 0x72, 0x8)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc4203d1170, 0x72, 0xc4201339d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc4203d1170, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc4203d1110, 0xc420282000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc420144130, 0xc420282000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc420336100, 0xc420282000, 0x1000, 0x1000, 0x30, 0xc420133b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc4200fa960)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc4200fa960, 0x1, 0x0, 0x1, 0x0, 0xc42050e000, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc420336100)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 254 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7fa82bf1cab0, 0x72, 0xd)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc4204628b0, 0x72, 0xc42055f9d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc4204628b0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc420462850, 0xc420383000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc42012a220, 0xc420383000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc4204ee600, 0xc420383000, 0x1000, 0x1000, 0x30, 0xc42055fb58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc42050f8c0)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc42050f8c0, 0x1, 0x0, 0x1, 0x0, 0xc4205a7740, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc4204ee600)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 129 [select, 9 minutes]:
net/http.(*persistConn).writeLoop(0xc42000b400)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 663 [select]:
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).runHeartbeat(0xc4202a2d80, 0xc4204eec00, 0xc4203ee900, 0xf, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:292 +0x650
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).GRPCDial.func1.1.1()
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:227 +0x69
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc42027c870, 0xc4203d9500)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 299 [select, 9 minutes]:
net/http.(*persistConn).writeLoop(0xc420502300)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 199 [chan receive, 9 minutes]:
database/sql.(*DB).connectionOpener(0xc4201711e0)
	/usr/local/go/src/database/sql/sql.go:730 +0x4a
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x1e9

goroutine 184 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7fa82bf1cdb0, 0x72, 0x9)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc42036a4c0, 0x72, 0xc4201379d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc42036a4c0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc42036a460, 0xc420560000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc42012a1a8, 0xc420560000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc4204ee300, 0xc420560000, 0x1000, 0x1000, 0x30, 0xc420137b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc4200fb500)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc4200fb500, 0x1, 0x0, 0x1, 0x0, 0xc42050e600, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc4204ee300)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 298 [IO wait, 9 minutes]:
net.runtime_pollWait(0x7fa82bf1c930, 0x72, 0xf)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420010ae0, 0x72, 0xc4205799d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420010ae0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc420010a80, 0xc420324000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc4201441b0, 0xc420324000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc420502300, 0xc420324000, 0x1000, 0x1000, 0x30, 0xc420579b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc4202ab1a0)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc4202ab1a0, 0x1, 0x0, 0x1, 0x0, 0xc4201ca9c0, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc420502300)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 481 [IO wait, 8 minutes]:
net.runtime_pollWait(0x7fa82bf1c1b0, 0x72, 0x19)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420209100, 0x72, 0xc4205779d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420209100, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc4202090a0, 0xc42039f000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc420144218, 0xc42039f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc4204eed00, 0xc42039f000, 0x1000, 0x1000, 0x30, 0xc420577b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc42028b740)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc42028b740, 0x1, 0x0, 0x1, 0x0, 0xc4204d32c0, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc4204eed00)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 370 [IO wait, 8 minutes]:
net.runtime_pollWait(0x7fa82bf1c6f0, 0x72, 0x12)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc4202d8c30, 0x72, 0xc4202c39d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc4202d8c30, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc4202d8bd0, 0xc4202d6000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc42012a238, 0xc4202d6000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc420502400, 0xc4202d6000, 0x1000, 0x1000, 0x30, 0xc4202c3b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc420327aa0)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc420327aa0, 0x1, 0x0, 0x1, 0x0, 0xc42050eea0, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc420502400)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 362 [IO wait, 8 minutes]:
net.runtime_pollWait(0x7fa82bf1c630, 0x72, 0x13)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420463170, 0x72, 0xc42055e9d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420463170, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc420463110, 0xc420506000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc420510108, 0xc420506000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc4204ee800, 0xc420506000, 0x1000, 0x1000, 0x30, 0xc42055eb58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc42050f800)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc42050f800, 0x1, 0x0, 0x1, 0x0, 0xc42014d800, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc4204ee800)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 267 [select, 9 minutes]:
net/http.(*persistConn).writeLoop(0xc420502900)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 303 [IO wait, 8 minutes]:
net.runtime_pollWait(0x7fa82bf1c7b0, 0x72, 0x11)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420011f00, 0x72, 0xc4202c59d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420011f00, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc420011ea0, 0xc420206000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc420510138, 0xc420206000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc4204ee500, 0xc420206000, 0x1000, 0x1000, 0x30, 0xc4202c5b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc420427800)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc420427800, 0x1, 0x0, 0x1, 0x0, 0xc4205c66c0, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc4204ee500)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 304 [select, 8 minutes]:
net/http.(*persistConn).writeLoop(0xc4204ee500)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 363 [select, 8 minutes]:
net/http.(*persistConn).writeLoop(0xc4204ee800)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 369 [IO wait, 8 minutes]:
net.runtime_pollWait(0x7fa82bf1c570, 0x72, 0x14)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc4204d6d80, 0x72, 0xc4205759d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc4204d6d80, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc4204d6d20, 0xc42033e000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc4204f20d0, 0xc42033e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc4204eeb00, 0xc42033e000, 0x1000, 0x1000, 0x30, 0xc420575b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc420219860)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc420219860, 0x1, 0x0, 0x1, 0x0, 0xc42026c720, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc4204eeb00)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 526 [select, 7 minutes]:
net/http.(*persistConn).writeLoop(0xc4204ee900)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 602 [select, 7 minutes]:
net/http.(*persistConn).readLoop(0xc420336300)
	/usr/local/go/src/net/http/transport.go:1541 +0x9c9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 402 [select, 8 minutes]:
net/http.(*persistConn).writeLoop(0xc4204eeb00)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 525 [IO wait, 7 minutes]:
net.runtime_pollWait(0x7fa82bf1c330, 0x72, 0x18)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420279410, 0x72, 0xc4201349d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420279410, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc4202793b0, 0xc4204e1000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc4204f20e8, 0xc4204e1000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc4204ee900, 0xc4204e1000, 0x1000, 0x1000, 0x30, 0xc420134b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc420327440)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc420327440, 0x1, 0x0, 0x1, 0x0, 0xc4203264e0, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc4204ee900)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 447 [IO wait, 8 minutes]:
net.runtime_pollWait(0x7fa82bf1c3f0, 0x72, 0x16)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420208ae0, 0x72, 0xc4202c49d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420208ae0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc420208a80, 0xc42026e000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc4204f2088, 0xc42026e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc420502700, 0xc42026e000, 0x1000, 0x1000, 0x30, 0xc4202c4b58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc420569140)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc420569140, 0x1, 0x0, 0x1, 0x0, 0xc420218b40, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc420502700)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 448 [select, 8 minutes]:
net/http.(*persistConn).writeLoop(0xc420502700)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 553 [IO wait, 7 minutes]:
net.runtime_pollWait(0x7fa82bf252b0, 0x72, 0x1a)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420278760, 0x72, 0xc42037b818, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420278760, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc420278700, 0xc420340000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc420144150, 0xc420340000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc420336300, 0xc420340000, 0x1000, 0x1000, 0x9cc868, 0xc4205298a8, 0x7d)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc4203270e0)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).ReadSlice(0xc4203270e0, 0x10a, 0x2, 0x0, 0x199, 0x186, 0x186)
	/usr/local/go/src/bufio/bufio.go:330 +0xb5
net/http/internal.readChunkLine(0xc4203270e0, 0x5d3659, 0xc420369d38, 0x0, 0x1, 0xc42037ba88)
	/usr/local/go/src/net/http/internal/chunked.go:110 +0x34
net/http/internal.(*chunkedReader).beginChunk(0xc4201d89c0)
	/usr/local/go/src/net/http/internal/chunked.go:47 +0x32
net/http/internal.(*chunkedReader).Read(0xc4201d89c0, 0xc420310001, 0x5ff, 0x5ff, 0x619db0, 0xc42037bb70, 0xc42037bb78)
	/usr/local/go/src/net/http/internal/chunked.go:77 +0x7b
net/http.(*body).readLocked(0xc4201407c0, 0xc420310001, 0x5ff, 0x5ff, 0x2, 0x2, 0x0)
	/usr/local/go/src/net/http/transfer.go:651 +0x61
net/http.(*body).Read(0xc4201407c0, 0xc420310001, 0x5ff, 0x5ff, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transfer.go:643 +0xf6
net/http.(*bodyEOFSignal).Read(0xc420140840, 0xc420310001, 0x5ff, 0x5ff, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1974 +0x116
encoding/json.(*Decoder).refill(0xc420529860, 0xa, 0x9)
	/usr/local/go/src/encoding/json/stream.go:152 +0xfa
encoding/json.(*Decoder).readValue(0xc420529860, 0x0, 0x0, 0x80)
	/usr/local/go/src/encoding/json/stream.go:128 +0x198
encoding/json.(*Decoder).Decode(0xc420529860, 0x13a1f80, 0xc420148100, 0x0, 0x0)
	/usr/local/go/src/encoding/json/stream.go:57 +0x8e
github.com/docker/docker/client.(*Client).Events.func1(0xc420556060, 0xc420278540, 0x0, 0x0, 0x0, 0x0, 0xc4201d83f0, 0x7fa82bf23670, 0xc4201b7080, 0xc420556000)
	/go/src/github.com/docker/docker/client/events.go:50 +0x2a7
created by github.com/docker/docker/client.(*Client).Events
	/go/src/github.com/docker/docker/client/events.go:63 +0xe2

goroutine 679 [chan receive, 7 minutes]:
github.com/cockroachdb/cockroach/pkg/rpc.NewContext.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:138 +0x76
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc42027c870, 0xc4203171a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:197 +0x66

goroutine 681 [select, 7 minutes]:
google.golang.org/grpc/transport.(*http2Client).controller(0xc420336500)
	/go/src/google.golang.org/grpc/transport/http2_client.go:999 +0x59d
created by google.golang.org/grpc/transport.newHTTP2Client
	/go/src/google.golang.org/grpc/transport/http2_client.go:233 +0xe78

goroutine 501 [IO wait, 8 minutes]:
net.runtime_pollWait(0x7fa82bf1c270, 0x72, 0x17)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420203330, 0x72, 0xc42003f9d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420203330, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc4202032d0, 0xc420341000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc4204f20f0, 0xc420341000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc420502b00, 0xc420341000, 0x1000, 0x1000, 0x30, 0xc42003fb58, 0x60033c)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc4204d3aa0)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc4204d3aa0, 0x1, 0x0, 0x1, 0x0, 0xc420326ae0, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc420502b00)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 502 [select, 8 minutes]:
net/http.(*persistConn).writeLoop(0xc420502b00)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 604 [select, 7 minutes]:
github.com/cockroachdb/cockroach/pkg/acceptance/cluster.(*LocalCluster).monitor.func1(0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/localcluster.go:608 +0x4aa
github.com/cockroachdb/cockroach/pkg/acceptance/cluster.(*LocalCluster).monitor(0xc420336200)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/localcluster.go:628 +0x61
created by github.com/cockroachdb/cockroach/pkg/acceptance/cluster.(*LocalCluster).Start
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/localcluster.go:652 +0x57e

goroutine 603 [select, 7 minutes]:
net/http.(*persistConn).writeLoop(0xc420336300)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 606 [IO wait, 7 minutes]:
net.runtime_pollWait(0x7fa82bf24d70, 0x72, 0x1b)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420153bf0, 0x72, 0xc4204ce9d0, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420153bf0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc420153b90, 0xc420541000, 0x1000, 0x1000, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc4204f20b0, 0xc420541000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
net/http.(*persistConn).Read(0xc42000a800, 0xc420541000, 0x1000, 0x1000, 0x8a0d50, 0xc4204ceb58, 0x5c90ad)
	/usr/local/go/src/net/http/transport.go:1261 +0x154
bufio.(*Reader).fill(0xc4204c6300)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Peek(0xc4204c6300, 0x1, 0xc4204cebbd, 0x1, 0x0, 0xc420369e60, 0x0)
	/usr/local/go/src/bufio/bufio.go:129 +0x62
net/http.(*persistConn).readLoop(0xc42000a800)
	/usr/local/go/src/net/http/transport.go:1418 +0x1a1
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1062 +0x4e9

goroutine 607 [select, 7 minutes]:
net/http.(*persistConn).writeLoop(0xc42000a800)
	/usr/local/go/src/net/http/transport.go:1646 +0x3bd
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1063 +0x50e

goroutine 680 [IO wait]:
net.runtime_pollWait(0x7fa82bf25670, 0x72, 0x1c)
	/usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc420279560, 0x72, 0xc420576828, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc420279560, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc420279500, 0xc4204e2c00, 0x400, 0x400, 0x0, 0x1e9e980, 0xc420014068)
	/usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc420144228, 0xc4204e2c00, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:173 +0x70
crypto/tls.(*block).readFromUntil(0xc420453560, 0x7fa82bf65ec0, 0xc420144228, 0x5, 0xc420144228, 0xc420348100)
	/usr/local/go/src/crypto/tls/conn.go:476 +0x91
crypto/tls.(*Conn).readRecord(0xc42016ee00, 0x16af417, 0xc42016ef08, 0xc4203efd40)
	/usr/local/go/src/crypto/tls/conn.go:578 +0xc4
crypto/tls.(*Conn).Read(0xc42016ee00, 0xc4204d8000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:1113 +0x116
bufio.(*Reader).fill(0xc4205c7380)
	/usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Read(0xc4205c7380, 0xc4201e0578, 0x9, 0x9, 0xc420576ca8, 0x619eb0, 0xc420358d80)
	/usr/local/go/src/bufio/bufio.go:209 +0x1bc
io.ReadAtLeast(0x1e97000, 0xc4205c7380, 0xc4201e0578, 0x9, 0x9, 0x9, 0xc4201d2460, 0xc4201d2458, 0x1e9adc0)
	/usr/local/go/src/io/io.go:307 +0xa4
io.ReadFull(0x1e97000, 0xc4205c7380, 0xc4201e0578, 0x9, 0x9, 0xc420576eb0, 0xee1f10, 0xc42028d2c0)
	/usr/local/go/src/io/io.go:325 +0x58
golang.org/x/net/http2.readFrameHeader(0xc4201e0578, 0x9, 0x9, 0x1e97000, 0xc4205c7380, 0x0, 0xc400000000, 0xc420348100, 0xc)
	/go/src/golang.org/x/net/http2/frame.go:237 +0x7b
golang.org/x/net/http2.(*Framer).ReadFrame(0xc4201e0540, 0x0, 0x0, 0x0, 0x0)
	/go/src/golang.org/x/net/http2/frame.go:464 +0xa4
google.golang.org/grpc/transport.(*framer).readFrame(0xc420453ec0, 0xc42017da10, 0xc42017da10, 0x0, 0x0)
	/go/src/google.golang.org/grpc/transport/http_util.go:505 +0x2f
google.golang.org/grpc/transport.(*http2Client).reader(0xc420336500)
	/go/src/google.golang.org/grpc/transport/http2_client.go:921 +0xb6
created by google.golang.org/grpc/transport.newHTTP2Client
	/go/src/google.golang.org/grpc/transport/http2_client.go:203 +0x8b1

goroutine 662 [select, 7 minutes]:
google.golang.org/grpc.(*addrConn).transportMonitor(0xc4201fc500)
	/go/src/google.golang.org/grpc/clientconn.go:720 +0x63e
google.golang.org/grpc.(*ClientConn).resetAddrConn.func1(0xc4201fc500)
	/go/src/google.golang.org/grpc/clientconn.go:479 +0x1de
created by google.golang.org/grpc.(*ClientConn).resetAddrConn
	/go/src/google.golang.org/grpc/clientconn.go:480 +0x3c7

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/explain.go
+++ b/pkg/sql/explain.go
@@ -113,6 +113,9 @@ func (p *planner) Explain(n *parser.Explain, autoCommit bool) (planNode, error) 
 			expanded: expanded,
 			results:  p.newContainerValuesNode(columns, 0),
 		}
+		// We want to show placeholder types, so ensure no values
+		// are missing.
+		p.semaCtx.Placeholders.FillUnassigned()
 		return node, nil
 
 	case explainPlan:

--- a/pkg/sql/expr_filter.go
+++ b/pkg/sql/expr_filter.go
@@ -1,5 +1,4 @@
 // Copyright 2016 The Cockroach Authors.
-
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -72,8 +71,9 @@ func (v *varConvertVisitor) VisitPre(expr parser.Expr) (recurse bool, newExpr pa
 	}
 
 	if varExpr, ok := expr.(parser.VariableExpr); ok {
-		// Ignore sub-queries
-		if _, isSubquery := expr.(*subquery); isSubquery {
+		// Ignore sub-queries and placeholders
+		switch expr.(type) {
+		case *subquery, *parser.Placeholder:
 			return false, expr
 		}
 

--- a/pkg/sql/parser/placeholders.go
+++ b/pkg/sql/parser/placeholders.go
@@ -16,7 +16,13 @@
 
 package parser
 
-import "fmt"
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/pkg/util"
+)
 
 // PlaceholderTypes relates placeholder names to their resolved type.
 type PlaceholderTypes map[string]Type
@@ -51,6 +57,34 @@ func (p *PlaceholderInfo) Assign(src *PlaceholderInfo) {
 	} else {
 		p.Clear()
 	}
+}
+
+// FillUnassigned sets all unsassigned placeholders to NULL.
+func (p *PlaceholderInfo) FillUnassigned() {
+	for pn := range p.Types {
+		if _, ok := p.Values[pn]; !ok {
+			p.Values[pn] = DNull
+		}
+	}
+}
+
+// AssertAllAssigned ensures that all placeholders that are used also
+// have a value assigned.
+func (p *PlaceholderInfo) AssertAllAssigned() error {
+	var missing []string
+	for pn := range p.Types {
+		if _, ok := p.Values[pn]; !ok {
+			missing = append(missing, "$"+pn)
+		}
+	}
+	if len(missing) > 0 {
+		sort.Strings(missing)
+		return fmt.Errorf("no value provided for placeholder%s: %s",
+			util.Pluralize(int64(len(missing))),
+			strings.Join(missing, ", "),
+		)
+	}
+	return nil
 }
 
 // Type returns the known type of a placeholder.

--- a/pkg/sql/pgwire_test.go
+++ b/pkg/sql/pgwire_test.go
@@ -371,6 +371,13 @@ func TestPGPreparedQuery(t *testing.T) {
 			baseTest.SetArgs(3.1).Error(`pq: error in argument for $1: strconv.ParseBool: parsing "3.1": invalid syntax`),
 			baseTest.SetArgs("").Error(`pq: error in argument for $1: strconv.ParseBool: parsing "": invalid syntax`),
 		},
+		"SELECT CASE 40+2 WHEN 42 THEN 51 ELSE $1::INT END": {
+			baseTest.Error(
+				"pq: no value provided for placeholder: $1",
+			).PreparedError(
+				"sql: statement expects 1 inputs; got 0",
+			),
+		},
 		"SELECT $1::int > $2::float": {
 			baseTest.SetArgs(2, 1).Results(true),
 			baseTest.SetArgs("2", 1).Results(true),
@@ -520,7 +527,7 @@ func TestPGPreparedQuery(t *testing.T) {
 		"SELECT a FROM d.T WHERE a = $1 AND (SELECT a >= $2 FROM d.T WHERE a = $1)": {
 			baseTest.SetArgs(10, 5).Results(10),
 			baseTest.Error(
-				"pq: no value provided for placeholder: $1",
+				"pq: no value provided for placeholders: $1, $2",
 			).PreparedError(
 				"sql: statement expects 2 inputs; got 0",
 			),

--- a/pkg/sql/pgwire_test.go
+++ b/pkg/sql/pgwire_test.go
@@ -302,6 +302,11 @@ type preparedQueryTest struct {
 	results [][]interface{}
 	others  int
 	error   string
+	// preparedError determines the error to expect upon stmt.Query()
+	// (executing a prepared statement), as opposed to db.Query()
+	// (direct query without prepare). If left empty, error above is
+	// used for both.
+	preparedError string
 }
 
 func (p preparedQueryTest) SetArgs(v ...interface{}) preparedQueryTest {
@@ -321,6 +326,11 @@ func (p preparedQueryTest) Others(o int) preparedQueryTest {
 
 func (p preparedQueryTest) Error(err string) preparedQueryTest {
 	p.error = err
+	return p
+}
+
+func (p preparedQueryTest) PreparedError(err string) preparedQueryTest {
+	p.preparedError = err
 	return p
 }
 
@@ -509,6 +519,11 @@ func TestPGPreparedQuery(t *testing.T) {
 		},
 		"SELECT a FROM d.T WHERE a = $1 AND (SELECT a >= $2 FROM d.T WHERE a = $1)": {
 			baseTest.SetArgs(10, 5).Results(10),
+			baseTest.Error(
+				"pq: no value provided for placeholder: $1",
+			).PreparedError(
+				"sql: statement expects 2 inputs; got 0",
+			),
 		},
 		"SELECT * FROM (VALUES (1), (2), (3), (4)) AS foo (a) LIMIT $1 OFFSET $2": {
 			baseTest.SetArgs(1, 0).Results(1),
@@ -570,7 +585,7 @@ func TestPGPreparedQuery(t *testing.T) {
 	}
 	defer db.Close()
 
-	runTests := func(query string, tests []preparedQueryTest, queryFunc func(...interface{}) (*gosql.Rows, error)) {
+	runTests := func(query string, prepared bool, tests []preparedQueryTest, queryFunc func(...interface{}) (*gosql.Rows, error)) {
 		for _, test := range tests {
 			if testing.Verbose() || log.V(1) {
 				log.Infof(context.Background(), "query: %s", query)
@@ -579,10 +594,16 @@ func TestPGPreparedQuery(t *testing.T) {
 			if err != nil {
 				if test.error == "" {
 					t.Errorf("%s: %v: unexpected error: %s", query, test.qargs, err)
-				} else if err.Error() != test.error {
-					t.Errorf("%s: %v: expected error: %s, got %s", query, test.qargs, test.error, err)
+				} else {
+					expectedErr := test.error
+					if prepared && test.preparedError != "" {
+						expectedErr = test.preparedError
+					}
+					if err.Error() != expectedErr {
+						t.Errorf("%s: %v: expected error: %s, got %s", query, test.qargs, expectedErr, err)
+					}
+					continue
 				}
-				continue
 			}
 			defer rows.Close()
 
@@ -656,7 +677,7 @@ CREATE TABLE d.str (s STRING, b BYTES);`
 	}
 
 	for query, tests := range queryTests {
-		runTests(query, tests, func(args ...interface{}) (*gosql.Rows, error) {
+		runTests(query, false, tests, func(args ...interface{}) (*gosql.Rows, error) {
 			return db.Query(query, args...)
 		})
 	}
@@ -668,7 +689,7 @@ CREATE TABLE d.str (s STRING, b BYTES);`
 			func() {
 				defer stmt.Close()
 
-				runTests(query, tests, stmt.Query)
+				runTests(query, true, tests, stmt.Query)
 			}()
 		}
 	}

--- a/pkg/sql/plan.go
+++ b/pkg/sql/plan.go
@@ -220,6 +220,9 @@ func (p *planner) makePlan(stmt parser.Statement, autoCommit bool) (planNode, er
 	if err != nil {
 		return nil, err
 	}
+	if err := p.semaCtx.Placeholders.AssertAllAssigned(); err != nil {
+		return nil, err
+	}
 	if err := plan.expandPlan(); err != nil {
 		return nil, err
 	}

